File: Selenium/Activities/Activity22.java
package Activities;
import io.github.bonigarcia.wdm.WebDriverManager;
import org.openqa.selenium.Alert;
import org.openqa.selenium.By;
import org.openqa.selenium.WebDriver;
import org.openqa.selenium.WebElement;
import org.openqa.selenium.firefox.FirefoxDriver;
import org.openqa.selenium.interactions.Actions;
import org.openqa.selenium.support.ui.ExpectedCondition;
import org.openqa.selenium.support.ui.ExpectedConditions;
import org.openqa.selenium.support.ui.Select;
import org.openqa.selenium.support.ui.WebDriverWait;

import java.time.Duration;
import java.util.List;

public class Activity22 {
    public static void main(String[] args) throws InterruptedException {
        WebDriverManager.firefoxdriver().setup();
        System.setProperty(FirefoxDriver.SystemProperty.BROWSER_LOGFILE, "/dev/null");
        WebDriver newDriver = new FirefoxDriver();
        WebDriverWait wait = new WebDriverWait(newDriver, Duration.ofSeconds(10));
        newDriver.get("https://www.training-support.net/selenium/popups");
        System.out.println("Title of page: " + newDriver.getTitle());
        WebElement toolEle = newDriver.findElement(By.className("orange"));
        Actions action = new Actions(newDriver);
        action.moveToElement(toolEle).pause(1000).build().perform();
        System.out.println("Message is: "+ toolEle.getAttribute("data-tooltip"));
        toolEle.click();
        newDriver.findElement(By.id("username")).sendKeys("admin");
        newDriver.findElement(By.id("password")).sendKeys("password");
        WebElement greenEle = newDriver.findElement(By.xpath("//button[text()='Log in']"));
        greenEle.click();
        System.out.println("Login message is: "+newDriver.findElement(By.id("action-confirmation")).getText());
        Thread.sleep(300);
        newDriver.quit();
    }
}

File: Selenium/Activities/Activity22.py
import time

from selenium import webdriver
from selenium.webdriver import ActionChains
from selenium.webdriver.common.by import By
from selenium.webdriver.support import expected_conditions
from selenium.webdriver.support.select import Select
from selenium.webdriver.support.wait import WebDriverWait
from webdriver_manager.firefox import GeckoDriverManager
from selenium.webdriver.firefox.service import Service as FirefoxService

serv = FirefoxService(GeckoDriverManager().install())

with webdriver.Firefox(service=serv) as driver:
    driver.get("https://www.training-support.net/selenium/popups")
    print("Title is: ", driver.title)
    action = ActionChains(driver)
    sign_in = driver.find_element(By.XPATH, "//button[contains(text(), 'Sign In')]")
    action.move_to_element(sign_in).perform()
    print("On hover message: ", sign_in.get_attribute("data-tooltip"))
    sign_in.click()
    username = driver.find_element(By.ID, "username")
    password = driver.find_element(By.ID, "password")
    username.send_keys("admin")
    password.send_keys("password")
    driver.find_element(By.XPATH, "/html/body/div[4]/div/div[2]/div/div/button").click()
    print("Confirmation message: ", driver.find_element(By.ID, "action-confirmation").text)
    driver.quit()
